util/argo/argo.go

```diff
@@ -123,6 +125,29 @@ func WaitForRefresh(ctx context.Context, appIf v1alpha1.ApplicationInterface, na
 	return nil, fmt.Errorf("application refresh deadline exceeded")
 }
 
+func TestRepo(repo *argoappv1.Repository) error {
```

Comment:
Is this package a God Object? Would it be good to extract repo related code?

Reply:
Yes, `argo` package was intended to keep very argocd specific utils. I would split it into sub-packages to make it clear that this is argocd specific. E.g. `argo/creds`, `argo/apps` etc. Do you mind If I do it in next PR?

Reply:
no hurry